website/src/e-commerce/src/search.js

```diff
@@ -0,0 +1,46 @@
+import algoliasearch from 'algoliasearch/lite';
+import instantsearch from 'instantsearch.js';
+import {
+  brands,
+  categories,
+  clearFilters,
+  configuration,
+  freeShipping,
+  products,
+  hitsPerPage,
+  pagination,
+  priceInput,
+  priceSlider,
+  ratings,
+  searchBox,
+  sortBy,
+} from './widgets';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+const search = instantsearch({
+  searchClient,
+  indexName: 'instant_search',
+  routing: true,
+});
+
+search.addWidgets([
+  configuration,
+  searchBox,
+  clearFilters,
+  brands,
+  categories,
+  priceInput,
+  priceSlider,
+  freeShipping,
+  ratings,
+  sortBy,
+  hitsPerPage,
+  products,
+  pagination,
+]);
+
+export default search;
```

Comment:
since this isn't an entry point, I wonder if it should be wrapped into a function 🤔

Reply:
You mean like `getSearch = () => ...` ?
Can you tell me more about this?
Does it have something to do with "entry point"?
In my last project, I had some issues exporting an object like that.
So I had to wrap it in a function but couldn't understand why I had to do it.

Reply:
if you'd import this twice for two search instances next to each other (weird, but not impossible), these would be the same instance. A similar remark can be made for the widgets themselves, which is why I was more a fan of not splitting it up

Reply:
I get what you mean. Speaking of splitting it up, if we're going to wrap `search` instance in a function, IMO we need to wrap all the widgets in function too. Otherwise, wrapping `search` only doesn't make any change. And I'm not sure if we need to do that.

Reply:
The reason why this search is in `search.js` and not directly in `app.js` is that `app.js` will grow with more JavaScript (UI features in #3802, sticky header is to come, etc.).

We can indeed export a `startSearch()` function and call it from `app.js`.